### PR TITLE
Added Speedy Labs LightFox Config

### PR DESCRIPTION
**Pull request by speedypotato**
Speedy Labs LightFox

Technically still requires GP2040-CE to invert the main 12 buttons, which is currently not implemented.

### CODE REVIEW

File: .github/workflows/cmake.yml
Comment (TheTrainGoes) [inline .github/workflows/cmake.yml:1]: This is less a comment for you and more a note for me:

This should be in alphabetical order, needs to be updated

File: configs/LightFox/BoardConfig.h [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+#define GPIO_PIN_25 GpioAction::BUTTON_PRESS_A1     // A1     | Guide  | Home    | PS       | 13     | ~      |
+#define GPIO_PIN_26 GpioAction::BUTTON_PRESS_A2     // A2     | ~      | Capture | ~        | 14     | ~      |
+
+// Setting GPIO pins to assigned by add-on
+//
+#define GPIO_PIN_02 GpioAction::ASSIGNED_TO_ADDON //i2c sda
+#define GPIO_PIN_03 GpioAction::ASSIGNED_TO_ADDON //i2c scl
+#define GPIO_PIN_29 GpioAction::ASSIGNED_TO_ADDON //rgb
+#define GPIO_PIN_04 GpioAction::ASSIGNED_TO_ADDON //focus
+// #define GPIO_PIN_07 GpioAction::ASSIGNED_TO_ADDON //SUSTAIN_DP_MODE_RS
```
Comment (TheTrainGoes) [inline configs/LightFox/BoardConfig.h:41]: Lines 41,42,43,44 can all be removed if they are being commented out.  I see that they appear in other places as well

**Review by TheTrainGoes — CHANGES_REQUESTED** [review]
Good start to things.  

I would like to wait until the device is finished and the code needed to run it has been merged into main.  

Are you planning to open source the device?

**Comment by speedypotato** [comment]
> Good start to things.
> 
> I would like to wait until the device is finished and the code needed to run it has been merged into main.
> 
> Are you planning to open source the device?

Controllers are closed source while they are being actively sold.  I release files when a model is no longer being sold/supported by me.

File: configs/LightFox/BoardConfig.h (hunk 1 above)
Comment (speedypotato) [reply configs/LightFox/BoardConfig.h]: Done

**Comment by TheTrainGoes** [comment]
> > Good start to things.
> > I would like to wait until the device is finished and the code needed to run it has been merged into main.
> > Are you planning to open source the device?
> 
> Controllers are closed source while they are being actively sold. I release files when a model is no longer being sold/supported by me.

This puts us in an odd spot.  

For now we have paused any new inclusion of configs while we work through new submission guidelines.  

Is there an ETA on the lifecycle for the product before its open sourced?

**Pull request closed without merging**